frontend/src/components/employees/DeleteModal.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import toastr from 'toastr';
+
+import Button from '@material-ui/core/Button';
+import Modal from '@material-ui/core/Modal';
+
+import Clear from '@material-ui/icons/Clear';
+import { StaffData } from '../../constants/employees';
+
+import { fetchStaff } from '../../features/staffSlice';
+import { URL, getAuthenticationToken } from '../../constants/global';
+import 'toastr/build/toastr.min.css';
+import employeesStyle from './employees.module.scss';
+
+interface Props {
+  isOpen: boolean;
+  staff: StaffData;
+  closeModalFunction: () => void;
+}
+
+const DeleteModal = ({ isOpen, staff, closeModalFunction }: Props) => {
+  const dispatch = useDispatch();
+
+  const deleteUser = ({ id, firstName, surname }: StaffData) => () => {
+    dispatch(
+      fetchStaff(getAuthenticationToken(), `${URL}/staff/remove?id=${id}`)
+    );
```

Comment:
Is it worth having a separate thunk for deleting a user? Or is it best to have one which takes parameters like this and can be used for different purposes?

Reply:
I agree that they should be separate. It feels strange for a a function whos job is to fetch something specific to accept a URL

Reply:
I can make the change suggested under #89 here if this works for you.

Reply:
I'm a little confused here. We're calling a method called fetchStaff and passing it a url of our choosing, which could in theory not be related to a staff endpoint at all? I think we should refactor that to be staff specific, or rename it.

One possible solution would be to create a generic 'apiService' class which has a 'get' method on it, which takes a url (which should be a const somewhere), and appends any headers required. This will also make it easier to append a token to every single request later on.

per my earlier point about the endpoint being a PUT rather than a GET, we'll presumably need another method here anyway, rather than calling fetchStaff.

Let me know your thoughts on this 👍🏻 I've shared my thoughts on #89 as well

Reply:
I admit, the naming was awful. I have refactored it now, although rather than a class, I have a bunch of functions.
What would the fields of this apiService class? Just the base URL?

Reply:
Functions are probably fine for now, and we could think about a class over in the AuthZ PR that's open.

I haven't given it much thought tbh - my basic thinking was having a class similar to this. Particularly note lines 38, and 62, which are our basic methods:

https://github.com/ScottLogic/mtg-app/blob/master/client/src/shared/network/NetworkService.ts